File: test_cases/c/gpt/01.c
#include <assert.h>
#include <math.h>

double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size);

int main(void) {
    const double EPS = 1e-9;

    {
        int nums1[] = {1, 3};
        int nums2[] = {2};
        double result = findMedianSortedArrays(nums1, 2, nums2, 1);
        assert(fabs(result - 2.0) < EPS);
    }

    {
        int nums1[] = {1, 2};
        int nums2[] = {3, 4};
        double result = findMedianSortedArrays(nums1, 2, nums2, 2);
        assert(fabs(result - 2.5) < EPS);
    }

    {
        int nums2[] = {1};
        double result = findMedianSortedArrays(NULL, 0, nums2, 1);
        assert(fabs(result - 1.0) < EPS);
    }

    {
        int nums1[] = {-5, -3, -1};
        int nums2[] = {-4, -2, 0};
        double result = findMedianSortedArrays(nums1, 3, nums2, 3);
        assert(fabs(result - (-2.5)) < EPS);
    }

    {
        int nums1[] = {-1000000, 1000000};
        int nums2[] = {-1000000, 1000000};
        double result = findMedianSortedArrays(nums1, 2, nums2, 2);
        assert(fabs(result - 0.0) < EPS);
    }

    return 0;
}

File: test_cases/python/gpt/01.py
def run_tests(solution_module):
    f = solution_module.findMedianSortedArrays
    assert abs(f([1, 3], [2]) - 2.0) < 1e-9
    assert abs(f([1, 2], [3, 4]) - 2.5) < 1e-9
    assert abs(f([], [1]) - 1.0) < 1e-9
    assert abs(f([-5, -3, -1], [-4, -2, 0]) - (-2.5)) < 1e-9
    assert abs(f([-1000000, 1000000], [-1000000, 1000000]) - 0.0) < 1e-9
